# HG changeset patch
# User Fabian Groffen <grobian@gentoo.org>
# Date 1464576396 -3600
#      Mon May 30 03:46:36 2016 +0100
# Node ID 61fdfc04fa6cca48644fde747bb4ece129fd815b
# Parent  58536ebf2669daab8332c262e5ae73ad9edf568d
[PATCH] change_folder_next
This patch brings back the original behaviour of changer-folder, which
some people find more useful.  It suggests the next folder in your
mailboxes list that has new mail measured from the current folder,
instead of the first folder from your mailboxes list in any case.  This
prevents starvation of folders defined at the back of your list ;)
See discussion: http://thread.gmane.org/gmane.mail.mutt.devel/12457
---
 PATCHES     | 1 +
 curs_main.c | 7 +++++++
 init.h      | 7 +++++++
 mutt.h      | 1 +
 4 files changed, 16 insertions(+)

diff -r 58536ebf2669 -r 61fdfc04fa6c PATCHES
--- a/PATCHES	Sat Mar 31 14:15:49 2018 +0200
+++ b/PATCHES	Mon May 30 03:46:36 2016 +0100
@@ -1,3 +1,4 @@
+patch-fg.change_folder_next-gentoo
 patch-vvv.quote-gentoo
 patch-xterm-no-tigetstr-on-tty-gentoo
 patch-xterm-only-window-title-gentoo
diff -r 58536ebf2669 -r 61fdfc04fa6c curs_main.c
--- a/curs_main.c	Sat Mar 31 14:15:49 2018 +0200
+++ b/curs_main.c	Mon May 30 03:46:36 2016 +0100
@@ -1836,7 +1836,14 @@ int mutt_index_menu (void)
 	  }
 	  else
 #endif
+	  {
+	  if (option (OPTCHANGEFOLDERNEXT) && Context && Context->path)
+	  {
+	    strfcpy (buf, Context->path, sizeof (buf));
+	    mutt_pretty_mailbox (buf, sizeof(buf));
+	  }
 	  mutt_buffy (buf, sizeof (buf));
+	  }
 
           if (mutt_enter_fname (cp, buf, sizeof (buf), 1) == -1)
           {
diff -r 58536ebf2669 -r 61fdfc04fa6c init.h
--- a/init.h	Sat Mar 31 14:15:49 2018 +0200
+++ b/init.h	Mon May 30 03:46:36 2016 +0100
@@ -417,6 +417,13 @@ struct option_t MuttVars[] = {
   ** mailbox by performing a fast mailbox scan when it is defined.
   ** Afterwards the new mail status is tracked by file size changes.
   */
+  { "change_folder_next", DT_BOOL, R_NONE, OPTCHANGEFOLDERNEXT, 0 },
+  /*
+  ** .pp
+  ** When this variable is set, the change-folder command will start at the
+  ** next folder after the current folder in your mailbox list instead of
+  ** starting at the first folder.
+  */
   { "check_new",	DT_BOOL, R_NONE, OPTCHECKNEW, 1 },
   /*
   ** .pp
diff -r 58536ebf2669 -r 61fdfc04fa6c mutt.h
--- a/mutt.h	Sat Mar 31 14:15:49 2018 +0200
+++ b/mutt.h	Mon May 30 03:46:36 2016 +0100
@@ -372,6 +372,7 @@ enum
   OPTBEEP,
   OPTBEEPNEW,
   OPTBOUNCEDELIVERED,
+  OPTCHANGEFOLDERNEXT,
   OPTBRAILLEFRIENDLY,
   OPTCHECKMBOXSIZE,
   OPTCHECKNEW,
